_targets/build.project.armv7a7-imx6ull

```diff
@@ -113,26 +127,47 @@ b_image_target() {
 	b_log "Creating image from $PREFIX_ROOTFS"
 
 	b_mkscript_user "${UPLOAD_SCRIPT[@]}"
-	b_mkscript_preinit
-
-	# Build production image
-	b_log "RAM Image"
-	b_kernel_image "$PREFIX_BOOT/phoenix.img"
-	b_syspage_gen "$PREFIX_BOOT/phoenix.img" "$PLO_SCRIPT_DIR/$NAME_USER_SCRIPT"
 
-	b_log "FLASH Image"
-	b_mkscript_user "${PRIMARY_SCRIPT[@]}"
-	b_kernel_image "$PREFIX_BOOT/phoenix-kernel.img"
-	b_syspage_gen "$PREFIX_BOOT/phoenix-kernel.img" "$PLO_SCRIPT_DIR/$NAME_USER_SCRIPT"
+	if [ "$KERNEL_PLO_BOOT" = "y" ]; then
+		b_log "Image will boot from PLO."
+
+		b_log "FLASH image"
+		b_mkscript_user "${PRIMARY_SCRIPT[@]}"
+		b_prod_image
+	else
+		# Parsed by syspagen for map and console commands.
+		b_mkscript_preinit
+
+		# Build production image
+		b_log "RAM Image"
+		b_kernel_image "$PREFIX_BOOT/phoenix.img"
+		b_syspage_gen "$PREFIX_BOOT/phoenix.img" "$PLO_SCRIPT_DIR/$NAME_USER_SCRIPT"
+
+		b_log "FLASH Image"
+		# This is a dirty workaround.
```

Comment:
IMHO not a valid comment as it's used in production :P. This is a workaround for targets not using PLO as a second stage bootloader on 6ull - and it's used to produce kernel image with required syspage apps embedded (not only uploading, this image resides on NAND and is treated as a single blob by BootROM)